Mage/src/main/java/mage/game/GameImpl.java

```diff
@@ -3763,4 +3763,123 @@ public String toString() {
+                        for (Target target : ability.getTargets()) {
+                            target.replaceMutatedTarget(permanentId, permanent.getId());
+                        }
+                        if (ability instanceof TriggeredAbility) {
+                            state.getTriggers().add((TriggeredAbility) ability, null, permanent);
+                        }
+                    }
+                    state.getContinuousEffects().replaceMutatedObjects(newPermanent.getId(), permanent.getId(), this);
+
+                    for (TriggeredAbility ability : state.getTriggers().values()) {
+                        if (ability instanceof ZoneChangeTriggeredAbility && ability.getSourceId().equals(permanent.getId())) {
+                            ((ZoneChangeTriggeredAbility) ability).addMutatedSourceId(permanent.getId());
+                            ((ZoneChangeTriggeredAbility) ability).addMutatedSourceId(newPermanent.getId());
+                        }
+                    }
+                    for (Ability ability : permanent.getAbilities()) {
+                        if (ability instanceof ZoneChangeTriggeredAbility) {
+                            ((ZoneChangeTriggeredAbility) ability).addMutatedSourceId(permanent.getId());
+                            ((ZoneChangeTriggeredAbility) ability).addMutatedSourceId(newPermanent.getId());
+                        }
+                    }
+
+                    fireEvent(GameEvent.getEvent(
+                            GameEvent.EventType.CREATURE_MUTATED, permanentId,
+                            source.getSpellAbility(), source.getControllerId()
+                    ));
+                    return true;
+                }
+            } else {
+                if (newPermanent.applyMutateOver(permanent, this)) {
+                    state.addPermanentToMutateZone(permanent);
+                    getBattlefield().removePermanent(permanentId);
+                    state.removePermanentFromMutateZone(newPermanent.getId(), Zone.BATTLEFIELD);
+                    getBattlefield().addPermanent(newPermanent);
+                    if (!source.isCopy()) {
+                        getState().updateZoneChangeCounter(permanentId);
+                    }
+
+                    // Replace any targets on the stack pointing to the old permanent to the new permanent
+                    for (Iterator<StackObject> it = state.getStack().iterator(); it.hasNext();) {
+                        StackObject stackObject = it.next();
+                        for (Target targets : stackObject.getStackAbility().getTargets()) {
+                            for (UUID target : targets.getTargets()) {
+                                if (target.equals(permanentId)) {
+                                    targets.replaceMutatedTarget(permanentId, newPermanent.getId());
+                                    break;
+                                }
```

Comment:
So as its written, it should be able to handle targeting the same creature multiple times, but I added some unit tests to address it and they are failing due to some lingering `zoneChangeCounter` issues. 

Those issues are lingering because zoneChangeCounters are basically like black magic to me and I don't understand why they are stored everywhere, how to know what they are/should be, and why it is that when I manipulate the zoneChangeCounters to fix these two new unit tests, it breaks 4 others...